Unit test

jonathan016 opened the pull request:
The previous PR for branch unit-test has the incorrect base, so it was closed.

Several basic unit tests were added in three modules: repository (AdminRepositoryTest), service (EntryPointServiceTest), and web-controller (EntryPointControllerTest).

Please note that several implementations of the written unit tests may be incorrect, especially those related to database connection using Fongo.

jonathan016 committed:
Added basic unit tests

Unit tests for repository, service, and web-controller modules are
added. This is only basic unit tests. However, the repository unit test
is doubted as how to correctly implement repository unit test is currently unknown.

jonathan016 committed:
Edited util method in login service test

Util method for generating simple employee data for login is edited to
be more flexible.

jonathan016 committed:
Merge branch 'development' of github.com:jonathan016/OASIS-backend into unit-test

jonathan016 committed:
Finished login feature unit test

Login feature unit tests are added in this test. Further review is
required.

jonathan016 committed:
Merge branch 'development' of github.com:jonathan016/OASIS-backend into unit-test

jonathan016 committed:
Merge branch 'development' of github.com:jonathan016/OASIS-backend into unit-test

jonathan016 committed:
Added more web-controller tests

More unit test for web controller are added. Further review is required.

jonathan016 committed:
Merge branch 'development' of github.com:jonathan016/OASIS-backend into unit-test

jonathan016 committed:
Renamed test classes

Several test classes are renamed to fix conflict caused when pulling
from development branch.

ensist commented on repository/pom.xml:
```diff
@@ -17,6 +17,64 @@
+        <dependency>
+            <groupId>org.mongodb</groupId>
+            <artifactId>mongo-java-driver</artifactId>
+            <version>3.8.0</version>
```
jgn define versi selain di pom parent ya

ensist commented on service/src/test/java/com/oasis/service/test/login/EntryPointServiceTest.java:
```diff
@@ -0,0 +1,323 @@
+            );
+            verify(roleDeterminer, times(1)).determineRole(notDeletedUsernames[ i ]);
+
+            verifyNoMoreInteractions(employeeUtilServiceApi);
```
semua verifyNoMoreInteractions taroh di method @After aja

jonathan016 committed:
Merge branch 'development' of github.com:jonathan016/OASIS-backend into unit-test

jonathan016 committed:
Fixed dependency problem on mongodb

Dependency problem on mongodb with mong-java-driver is fixed in this
commit.

jonathan016 committed:
Merge branch 'development' of github.com:jonathan016/OASIS-backend into unit-test

jonathan016 committed:
Merge branch 'development' of github.com:jonathan016/OASIS-backend into unit-test

jonathan016 committed:
Initialized test classes & added several tests

Test classes for classes in modules repository, service, and
web-controller are added. Several tests for
LastUniqueIdentifierRepository are added. Further review is required.

jonathan016 committed:
Added supervision repository unit tests

Several unit tests for supervision repository are added in this commit.
Only basic unit tests are added as time runs short. Further unit tests
are required.

jonathan016 committed:
Refactored unit tests, added more unit tests

Unit tests were refactored and more unit tests are added. Unit test on
service module is currently in progress.

jonathan016 committed:
Added unit test for active component manager

Unit tests for active component manager are added.

jonathan016 committed:
Added more unit tests

Several unit tests for dashboard service are added in this commit. Many
other unit tests are still required.

jonathan016 committed:
Merge branch 'development' of github.com:jonathan016/OASIS-backend into unit-test

jonathan016 commented:
Merging this branch with branch development for easier maintenance/development.

Comments from the reviewer of the pull request (Mr. Endri) have been resolved in the commits.
